src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -339,6 +369,36 @@ def test_process_declarations(f2008_parser):
     assert fake_parent.symbol_table.lookup("l5").shape == [2]
     assert fake_parent.symbol_table.lookup("l6").shape == [3]
 
+    # Initialisations of static constant values (parameters)
+    reader = FortranStringReader("integer, parameter :: i1 = 1")
+    fparser2spec = Specification_Part(reader).content[0]
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert fake_parent.symbol_table.lookup("i1").constant_value == 1
+
+    reader = FortranStringReader("real, parameter :: i2 = 2.2, i3 = 3.3")
+    fparser2spec = Specification_Part(reader).content[0]
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert fake_parent.symbol_table.lookup("i2").constant_value == 2.2
+    assert fake_parent.symbol_table.lookup("i3").constant_value == 3.3
+
+    # Static constant expresions are not supported
+    reader = FortranStringReader("real, parameter :: a = 1.1, b = a * 2")
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert "Could not process " in str(error.value)
+    assert "Initialisations with static expressions are not supported." \
+        in str(error.value)
+
+    # Static constant expresions are not supported
```

Comment:
Not at the moment, this is because right now I just set the datasymbols `constant_value` to the RHS value, which it implies a parameter.

For variable initialization there are some possible solutions: 1) insert an assignment at the beginning of the schedule, 2) add a new attribute `initial_value` for datasymbols. I think this and basic static expressions happen often enough that we should implement it, but maybe as part of a follow up PR?

Reply:
Yes, I agree. I think I was just suggesting that you re-word the comment slightly - apologies for being unclear. I've had to re-read it several times and now I can see what I meant was that this comment is a repeat of the one above when in fact it should say "initialisations are only supported for parameters" or some such.